### PR TITLE
[Feature] Implement the conformer backbone.

### PR DESCRIPTION
**Pull request by pengzhiliang**
## Motivation

Implement the backbone, [conformer](https://arxiv.org/abs/2105.03889), published in ICCV21.

And its official implementation in https://github.com/pengzhiliang/Conformer

## Modification

Add a new backbone `conformer.py` in `mmcls/backbone` , a new head `conformer_head.py` in  `mmcls/head`, and some config files in `configs`

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

Add a new backbone

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by mzr1996** [comment]
Hello, we have a [Contributing guide](https://mmclassification.readthedocs.io/en/latest/community/CONTRIBUTING.html), and please fix the lint issues according to this guide. Thanks~

File: mmcls/models/backbones/conformer.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+                 base_channel=64,
+                 mlp_ratio=4.,
+                 qkv_bias=True,
+                 cls_token=True,
+                 drop_path_rate=0.,
+                 norm_eval=True,
+                 frozen_stages=0,
+                 out_indices=(12, ),
+                 norm_cfg=dict(),
+                 stage_cfgs=dict(),
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply mmcls/models/backbones/conformer.py:321]: Well, have you finished it? I cannot find the modification :confused:

[the remark this answers is not in the document]
Comment (pengzhiliang) [reply mmcls/models/backbones/conformer.py]: Sorry, I ignored this.It's complete now.

**Review by mzr1996 — APPROVED** [review]
LGTM

**Pull request merged**